File: 2785. Sort Vowels in a String/Solution.py
class Solution:
    def sortVowels(self, s: str) -> str:
        vowels = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}
        vs = []
        for c in s:
            if c in vowels: vs.append(c)
        vs.sort(key = lambda x: ord(x))
        i = 0
        res = []
        for c in s:
            if c not in vowels:
                res.append(c)
            else:
                res.append(vs[i])
                i += 1
        return "".join(res)

File: 2785. Sort Vowels in a String/Solution.java
class Solution {
    public String sortVowels(String s) {
        Set<Character> vowels = new HashSet<>(List.of('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'));
        List<Character> vs = new ArrayList<>();
        List<Integer> ps = new ArrayList<>();
        char[] ss = s.toCharArray();
        for (int i = 0; i < ss.length; i++) {
            if (vowels.contains(ss[i])) {
                vs.add(ss[i]);
                ps.add(i);
            }
        }
        vs.sort((a, b) -> a - b);
        int i = 0;
        for (int p : ps) {
            ss[p] = vs.get(i);
            i++;
        }
        return new String(ss);
    }
}
